subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultTransformer.java

```diff
@@ -156,8 +164,8 @@ public HashCode getSecondaryInputHash() {
     }
 
     @Override
-    public ImmutableList<File> transform(File inputArtifact, File outputDir, ArtifactTransformDependencies dependencies) {
-        TransformAction transformAction = newTransformAction(inputArtifact, dependencies);
+    public ImmutableList<File> transform(File inputArtifact, File outputDir, ArtifactTransformDependencies dependencies, IncrementalInputs incrementalInputs) {
```

Comment:
Currently, it is not `@Nullable`. We need to know if the execution is incremental or not, even before we pass incremental inputs to the `Transformer`. We can find that out differently, so we could look it up. Given that the incremental inputs object is only created when the tranform action is executing, do you think it would make a big difference to look it up instead of creating it beforehand?